### DISCUSSION TITLE
Llamafile produces speed gains of 30% to 500% on CPU

### DISCUSSION THREAD

**Discussion by mictadlo** [open-webui/open-webui#4532] [General]
Hi,
There is [talk](https://www.youtube.com/watch?v=-mRi-B3t6fA)  on how  Llamafile produces speed gains of 30% to 500% on CPU.

How is it possible to combine llamafile with open-webui ?

Best wishes,

Michal

**Comment by ChristianWeyer** [comment]
Would be great to see an integration of the two, agreed!

**Comment by ChristianWeyer** [comment]
Have you found a way @mictadlo ?

**Comment by wirthual** [comment]
Hi @ChristianWeyer,

To use llamafile locally with open-webui I did the following steps:
1. Download a llama file you can run e.g. https://huggingface.co/wirthual/Meta-Llama-3.2-1B-Instruct-llamafile
2. Make the file executable and run the server on port 8082
```
wget https://huggingface.co/wirthual/Meta-Llama-3.2-1B-Instruct-llamafile/resolve/main/Meta-Llama-3.2-1B-Instruct-Q4_K_M.llamafile
chmod +x Meta-Llama-3.2-1B-Instruct-Q4_K_M.llamafile
./Meta-Llama-3.2-1B-Instruct-Q4_K_M.llamafile --port 8082 --nobrowser
```
This will start the llamafile server we are going to connect to using open-webui. Its important to keep this server running and not close the window ;)

3. Install open-webui using pip  (python 3.11 and I use a virtualenv here)
4. Launch openwebui and point it to the llamafile server
```
python3.11 -m venv .venv
source .venv/bin/activate
pip3 install open-webui
 ENABLE_OPENAI_API="true" OPENAI_API_BASE_URL="http://localhost:8082/v1" OPENAI_API_KEY="test" open-webui serve
```

Now open a browser and browse to [http://0.0.0.0:8080/](http://0.0.0.0:8080/) and you should be able to chat with your llamefile as show here:

https://github.com/user-attachments/assets/ad38ce08-528e-4315-a312-841ac48d1b11
